src/features/workspace/components/workspace-name.tsx

```diff
@@ -29,17 +29,24 @@ export function WorkspaceName({
   const formState = useFormState({
     workspaceName,
   });
-  const { values, updateFormValues } = formState;
+  const { values, updateFormValues, setInitialValues } = formState;
   const isDefault = workspaceName === "default";
   const isUneditable = isArchived || isPending || isDefault;
 
+  useEffect(() => {
+    setInitialValues({ workspaceName });
+  }, [setInitialValues, workspaceName]);
+
   const handleSubmit = (event: FormEvent) => {
     event.preventDefault();
 
     mutateAsync(
       { body: { name: workspaceName, rename_to: values.workspaceName } },
       {
-        onSuccess: () => navigate(`/workspace/${values.workspaceName}`),
+        onSuccess: () => {
+          formState.setInitialValues({ workspaceName: values.workspaceName });
+          navigate(`/workspace/${values.workspaceName}`);
```

Comment:
I have a doubt, if we navigate to a new route, why is it necessary to set the initial values again?

Reply:
yep cause the form is not reset automatically...I would like to add a library honestly to handle the form state...IMO it will be much simpler. During the v3 I am going to add react-hook-form